python/tchannel/tornado/peer.py

```diff
@@ -458,35 +489,110 @@ def send(self, arg1, arg2, arg3, traceflag=False, headers=None):
             parent_span_id = None
             trace_id = None
 
-        # set 'as' to 'raw if 'as' is empty
+        # set default transport headers
         headers = headers or {}
-        if 'as' not in headers:
-            headers['as'] = 'raw'
-
-        connection = yield self.peer.connect()
-        message_id = connection.next_message_id()
-        response = yield connection.send_request(
-            Request(
-                service=self.service,
-                argstreams=[InMemStream(endpoint), arg2, arg3],
-                id=message_id,
-                headers=headers,
-                tracing=Trace(
-                    name=endpoint,
-                    trace_id=trace_id,
-                    parent_span_id=parent_span_id,
-                    endpoint=Endpoint(self.peer.host,
-                                      self.peer.port,
-                                      self.service),
-                    traceflags=traceflag,
-                )
+        real_header = {}
+        for h in transport_headers:
+            real_header[h] = headers.get(h, None) or self.headers.get(h, None)
```

Comment:
1. TCP doesn't have a concept of transport headers.
2. HTTP has support for arbitrary headers. Although it doesn't play well with the application/protocol headers analogy unless you add the assumption that the application headers and request are part of the HTTP request body.
3. TChannel has a concept of separate application and transport headers. I shouldn't have to update my library just to pass in a new transport header. Further, I should be able to add my own customizations based on new transport headers if I so desire. The low level API should provide a way to specify the headers.
4. I don't know how many times I need to repeat this: If there is a need in the high level API to limit the protocol headers to a specific predefined set of keys, your API should not be accepting a dictionary. It should be accepting actual separate keyword arguments. If I pass in a dictionary to an API that says "headers", I expect **all** key-value pairs from the dictionary to be present in the outgoing packet. Ignoring items in my dictionary silently is wrong behavior.